### PR TITLE
add 'force_basic_auth' option

### PR DESCRIPTION
**Pull request by rule88**
by default "force_basic_auth" option is set to False, with this adjustment it is up to the user if it wants to enforce basic authentication.

##### SUMMARY
Add support to the URL lookup plugin to support basic authentication (option: force_basic_auth)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugin --> lookup --> url

##### ADDITIONAL INFORMATION
This change is will help people needing to do an URL lookup with basic authentication enabled. By default it still is disabled, but with the force_basic_auth switch (boolean) it can be enabled.

### CODE REVIEW

File: lib/ansible/plugins/lookup/url.py [hunk 1]
```diff
@@ -39,6 +39,10 @@
     type: dictionary
     default: {}
     version_added: "2.9"
+  force_basic_auth:
+    description: Force basic authentication
+    type: boolean
+    version_added: "2.9"
```
Comment (Akasurde) [inline lib/ansible/plugins/lookup/url.py]:
```suggestion
    version_added: "2.10"
    default: False
```

Since 2.9 is Out.

**Comment by Akasurde** [comment]
cc @sivel

**Review by Akasurde — DISMISSED** [review]

**Comment by rule88** [comment]
@Akasurde already committed, but is it wise to have the default declared "again" in this plugin? It seems to already have a default in its module_utils:  https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py

**Comment by samdoran** [comment]
This is ok, but adding options to the lookup one at a time will get out of hand. It would be better to go ahead and bring in all the options so the `uri` module has the same feature parity as the `url` lookup.

**Comment by rule88** [comment]
@samdoran - added most "basic" parameters that are also to be found in the underlying used lib/ansible/module_utils/urls.py library:

- force
- timeout
- http_agent
- follow_redirects
- use_gssapi
- unix_socket
- ca_path
- unredirected_headers

 Left out the parameters where the datatypes don't seem to be ever passed by Ansible as Ansible afaik doesn't support passing those types. They also don't seem to add value to the "lookup" feature.

- data
- method ( out of this PR as this seems to not add value in a "lookup" module, 
- last_mod_time
- client_cert
- client_key
- cookies

**Review by samdoran — CHANGES_REQUESTED** [review]
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Comment by rule88** [comment]
@samdoran - changelog fragment added!

**Review by samdoran — APPROVED** [review]

File: changelogs/fragments/64892-add-parameters-to-url_lookup_plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - parameters - add parameters to url lookup plugin as they also exist in module_utils (urls.py).
```
Comment (samdoran) [inline changelogs/fragments/64892-add-parameters-to-url_lookup_plugin.yaml]:
```suggestion
  - url lookup plugin - add parameters to match what is available in ``module_utils/urls.py``
```

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
rebuild_merge

**Pull request merged**